voice/src/main/kotlin/handlers/VoiceUpdateEventHandler.kt

```diff
@@ -0,0 +1,34 @@
+@file:OptIn(KordVoice::class)
+
+package dev.kord.voice.handlers
+
+import dev.kord.common.annotation.KordVoice
+import dev.kord.gateway.VoiceServerUpdate
+import dev.kord.voice.VoiceConnection
+import kotlinx.coroutines.flow.Flow
+import mu.KotlinLogging
+import dev.kord.gateway.Event as GatewayEvent
+
+private val voiceUpdateLogger = KotlinLogging.logger { }
+
+internal class VoiceUpdateEventHandler(
+    private val connection: VoiceConnection,
+    flow: Flow<GatewayEvent>,
+) : EventHandler<GatewayEvent>(flow, "VoiceUpdateInterceptor") {
+    override fun start() {
+        on<VoiceServerUpdate> { voiceServerUpdate ->
+            // see if this voice server update interests this voice connection
+            if (voiceServerUpdate.voiceServerUpdateData.guildId != connection.data.guildId) return@on
```

Comment:
move this into a variable name called isRelatedToConnection or some meaningful variable names rather than commenting what it does if possible.